sockeye/inference.py

```diff
@@ -423,6 +427,11 @@ def load_models(context: mx.context.Context,
         utils.check_condition(vocab.are_identical(*[source_vocabs[i][fi] for i in range(len(source_vocabs))]),
                               "Source vocabulary ids do not match. Factor %d" % fi)
 
+    source_with_eos = models[0].source_with_eos
+    utils.check_condition(all(source_with_eos == m.source_with_eos for m in models),
+                          "All models must match either take an additional EOS symbol on the source side or not. "
```

Comment:
grammar, maybe: "All models must agree on using source-side EOS symbols or not"

Reply:
sounds good, changed.